src/main/java/wolox/training/models/User.java

```diff
@@ -0,0 +1,194 @@
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public LocalDate getBirthdate() {
+        return birthdate;
+    }
+
+    public void setBirthdate(LocalDate birthdate) {
+        this.birthdate = birthdate;
+    }
+
+    public List<Book> getLibrary() {
+        return Collections.unmodifiableList(library);
+    }
+
+    public void setLibrary(List<Book> books) {
+        this.library = books;
+    }
+
+    /**
+     * This method is used to add a {@link Book} from the library collection of an {@link User}
+     *
+     * @param book: Book model class with the following attributes: (Book)
+     *     id: Id of the book (Long),
+     *     genre: Genre of the book (String),
+     *     author: Author of the book (String),
+     *     image: Image of the book (String),
+     *     title: Title of the book (String),
+     *     subtitle: Subtitle of the book (String),
+     *     publisher: Publisher of the book (String),
+     *     year: Year of the book (String),
+     *     pages: Pages of the book (Integer),
+     *     isbn: Isbn of the book (String).
+     * @exception BookAlreadyOwnException: throw a {@link BookAlreadyOwnException} in case that the book is already in the library collection
+     */
+    public void addBookToUser(Book book) {
+        if (book == null) {
+            throw new NullPointerException();
+        }
+        if (validateBook(book)) {
+            throw new BookAlreadyOwnException("Book is already in the user collection");
+        }
+        this.library.add(book);
+    }
+
+    /**
+     * This method is used to remove a {@link Book} from the library collection of an {@link User}.
+     *
+     * @param book: Book model class with the following attributes: (Book)
+     *     id: Id of the book (Long),
+     *     genre: Genre of the book (String),
+     *     author: Author of the book (String),
+     *     image: Image of the book (String),
+     *     title: Title of the book (String),
+     *     subtitle: Subtitle of the book (String),
+     *     publisher: Publisher of the book (String),
+     *     year: Year of the book (String),
+     *     pages: Pages of the book (Integer),
+     *     isbn: Isbn of the book (String).
+     */
+    public void removeBookToUser(Book book) {
+        if(this.library.contains(book)) {
+            this.library.remove(book);
+        }
+    }
+
+    /**
+     * This method is used to remove a {@link Book}  by id from the library collection of an {@link User}.
+     *
+     * @param id: Id of the book who is going to be removed from the library collection of the {@link User},
+     */
+    public void removeBookToUserById(Long id) {
+        for (Book b : this.library) {
+            if (b.getId().equals(id)) {
+                this.library.remove(b);
+            }
+        }
+    }
```

Comment:
This method should not be included in the User model, you need to pass the book directly, so the method you added above this one should suffice 😄

Reply:
Oki, perfect, thank you so much :smile_cat: